app/controllers/ClaimPeriodStartController.scala

```diff
@@ -0,0 +1,76 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import forms.ClaimPeriodStartFormProvider
+import javax.inject.Inject
+import models.{Mode, NormalMode, UserAnswers}
+import navigation.Navigator
+import pages.ClaimPeriodStartPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.ClaimPeriodStartView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ClaimPeriodStartController @Inject()(
+                                        override val messagesApi: MessagesApi,
+                                        sessionRepository: SessionRepository,
+                                        navigator: Navigator,
+                                        identify: IdentifierAction,
+                                        getData: DataRetrievalAction,
+                                        requireData: DataRequiredAction,
+                                        formProvider: ClaimPeriodStartFormProvider,
+                                        val controllerComponents: MessagesControllerComponents,
+                                        view: ClaimPeriodStartView
+                                      )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  private def form = formProvider()
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData) {
+    implicit request =>
+
+      val userAnswers = request.userAnswers
+        .getOrElse(UserAnswers(request.identifier))
+        .get(ClaimPeriodStartPage)
+
+      val preparedForm = userAnswers match {
+        case Some(date) => form.fill(date)
+        case None       => form
+      }
+
+      Ok(view(preparedForm))
+  }
+
+  def onSubmit: Action[AnyContent] = (identify andThen getData).async {
+    implicit request =>
+
+      form.bindFromRequest().fold(
+        formWithErrors =>
+          Future.successful(BadRequest(view(formWithErrors))),
+
+        value =>
+          for {
+            updatedAnswers <- Future.fromTry(request.userAnswers.getOrElse(UserAnswers(request.identifier)).set(ClaimPeriodStartPage, value))
+            _              <- sessionRepository.set(updatedAnswers)
+          } yield Redirect(navigator.nextPage(ClaimPeriodStartPage, NormalMode, updatedAnswers))
```

Comment:
what is `NormalMode`? what does that do?

Reply:
we have NormalMode for normal journey. and CheckMode for check-your-answers journey .. but we dont deal with check-your-answers journey as of now ... so we will only deal with NormalMode.... 
Scaffold generates these Modes based logic